Framework/Kernel/src/UserStringParser.cpp

```diff
@@ -149,14 +148,13 @@ void UserStringParser::Tokenize(const std::string &input,
                                 const std::string &delimiter,
                                 unsigned int &start, unsigned int &end,
                                 unsigned int &step) {
-  typedef boost::tokenizer<boost::char_separator<char>> tokenizer;
+  typedef Mantid::Kernel::StringTokenizer tokenizer;
 
-  boost::char_separator<char> seps(delimiter.c_str());
+  const std::string seps(delimiter.c_str());
   std::vector<std::string> temp;
   tokenizer tokens(input, seps);
-  for (tokenizer::const_iterator tokItr = tokens.begin();
-       tokItr != tokens.end(); ++tokItr) {
-    temp.push_back(*tokItr);
+  for (const auto &tokItr : tokens) {
+    temp.push_back(tokItr);
   }
```

Comment:
I think you can remove `temp` and use `tokens.asVector()` to pass a const reference.